test/NuGet.Core.Tests/NuGet.Packaging.Test/SigningTests/ChainBuilding/RetriableX509ChainBuildPolicyTests.cs

```diff
@@ -0,0 +1,133 @@
+        [InlineData(3, 700)]
+        public void Constructor_WhenArgumentsAreValid_InitializesProperties(
+            int expectedRetryCount,
+            int expectedMilliseconds)
+        {
+            TimeSpan expectedSleepInterval = TimeSpan.FromSeconds(expectedMilliseconds);
+
+            var policy = new RetriableX509ChainBuildPolicy(expectedRetryCount, expectedSleepInterval);
+
+            Assert.Equal(expectedRetryCount, policy.RetryCount);
+            Assert.Equal(expectedSleepInterval, policy.SleepInterval);
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public void Build_WhenBuildIsNull_Throws()
+        {
+            var policy = new RetriableX509ChainBuildPolicy(retryCount: 1, TimeSpan.Zero);
+
+            using (X509Certificate2 certificate = _fixture.GetDefaultCertificate())
+            {
+                ArgumentNullException exception = Assert.Throws<ArgumentNullException>(
+                    () => policy.Build(chain: null, certificate));
+
+                Assert.Equal("chain", exception.ParamName);
+            }
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public void Create_WhenCertificateIsNull_Throws()
+        {
+            var policy = new RetriableX509ChainBuildPolicy(retryCount: 1, TimeSpan.Zero);
+
+            using (var chain = new X509Chain())
+            {
+                ArgumentNullException exception = Assert.Throws<ArgumentNullException>(
+                    () => policy.Build(chain, certificate: null));
+
+                Assert.Equal("certificate", exception.ParamName);
+            }
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public void Build_WhenArgumentsAreValidAndBuildSucceeds_DoesNotRetry()
+        {
+            var policy = new RetriableX509ChainBuildPolicy(retryCount: 3, TimeSpan.FromSeconds(10));
+
+            using (var chain = new X509Chain())
+            using (X509Certificate2 certificate = GetTrustedCertificate())
+            {
+                Stopwatch stopwatch = Stopwatch.StartNew();
+
+                bool actualResult = policy.Build(chain, certificate);
+
+                Assert.True(actualResult);
+                Assert.True(stopwatch.Elapsed < policy.SleepInterval);
+            }
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public void Build_WhenArgumentsAreValidAndBuildFailureIsRetriable_Retries()
+        {
+            var policy = new RetriableX509ChainBuildPolicy(retryCount: 3, TimeSpan.FromMilliseconds(50));
+
+            using (var chain = new X509Chain())
+            using (X509Certificate2 certificate = _fixture.GetDefaultCertificate())
+            {
+                Stopwatch stopwatch = Stopwatch.StartNew();
+
+                bool actualResult = policy.Build(chain, certificate);
+
+                Assert.False(actualResult);
+                Assert.True(stopwatch.Elapsed > TimeSpan.FromMilliseconds(policy.RetryCount * policy.SleepInterval.TotalMilliseconds));
```

Comment:
Is there ever a case when this could take this long without actually retrying?  I don't generally trust unit tests that count on timing but I do not know how this certificate stuff works either, so maybe its an unwarranted fear.